server/ARCHITECTURE.md

```diff
@@ -0,0 +1,73 @@
+# Architecture
+
+This document describes the high-level architecture of Waterpark. If you want to familiarize yourself
+with the codebase, you are just in the right place!
+
+### Bird's Eye View
+
+![](https://blog.cleancoder.com/uncle-bob/images/2012-08-13-the-clean-architecture/CleanArchitecture.jpg)
+
+Waterpark's architecture is heavily modelled with Domain Driven Design. DDD is a strong tool when modelling complex and rich domains. You can learn more about Loo Labs' motivations for DDD [here]().
```

Comment:
`[here]()` is missing a link

Reply:
I left this as a placeholder to a a future Loo Labs' manifest/philosophy on providing a resource for students to learn clean software architecture and best practices. I'm thinking that document can live on our website. You think it's better if I remove this sentence in this PR?